homeassistant/components/fritzbox_callmonitor/__init__.py

```diff
@@ -1 +1,87 @@
-"""The fritzbox_callmonitor component."""
+"""The fritzbox_callmonitor integration."""
+from asyncio import gather
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+from requests.exceptions import ConnectionError as RequestsConnectionError
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .base import FritzBoxPhonebook
+from .const import (
+    CONF_PHONEBOOK,
+    CONF_PREFIXES,
+    DOMAIN,
+    FRITZ_BOX_PHONEBOOK_OBJECT,
+    PLATFORMS,
+    UNDO_UPDATE_LISTENER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the fritzbox_callmonitor integration."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up the fritzbox_callmonitor platforms."""
+    phonebook = FritzBoxPhonebook(
+        host=config_entry.data[CONF_HOST],
+        username=config_entry.data[CONF_USERNAME],
+        password=config_entry.data[CONF_PASSWORD],
+        phonebook_id=config_entry.data[CONF_PHONEBOOK],
+        prefixes=config_entry.options.get(CONF_PREFIXES),
+    )
+
+    try:
+        await hass.async_add_executor_job(phonebook.init_phonebook)
+    except (
+        RequestsConnectionError,
+        FritzSecurityError,
```

Comment:
Is this thrown when user/pass is invalid? If so, we should not raise `ConfigEntryNotReady` because the credentials will remain invalid.

Reply:
No, `FritzSecurityError` is thrown when the given user does not have the correct permissions to access the phonebook and the device settings of the FritzBox. Only `FritzConnectionException` when the credentials are invalid.

Reply:
Correct. `ConfigEntryNotReady` indicates that it's currently not working, but it can work in the future. For example, when the device is currently offline. 

If permissions or auth are incorrect, we should not retry, as we know that it will always fail. In that case `return False`.

Reply:
Fixed in https://github.com/home-assistant/core/pull/40736/commits/092bc37df8f72144274c10b28e35fea9b8f32e95